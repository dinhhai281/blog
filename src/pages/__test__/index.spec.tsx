import { render } from '@utils/testing';
import React from 'react';

import Home, { HomeProps } from '@pages/index';

describe('Home', () => {
  let props: HomeProps;
  beforeEach(() => {
    props = {
      data: {
        site: {
          siteMetadata: {
            author: 'author',
          },
        },
        allGithubData: {
          edges: [
            {
              node: {
                data: {
                  viewer: {
                    name: 'test',
                    avatarUrl: 'testurl',
                  },
                },
              },
            },
          ],
        },
      },
    };
  });

  it('render correctly', () => {
    const { asFragment } = render(<Home data={props.data}></Home>);

    expect(asFragment()).toMatchSnapshot();
  });
});
