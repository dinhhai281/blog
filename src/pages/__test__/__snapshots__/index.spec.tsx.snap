// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home render correctly 1`] = `
<DocumentFragment>
  <div
    class="container px-5 pt-8 mx-auto"
  >
    <header
      class="flex justify-between items-center mb-8"
    >
      <h1
        class="text-pink-700 text-4xl font-bold leading-10"
      >
        <a
          href="/en/about/"
        >
          Onyxia
        </a>
      </h1>
      <div>
         
      </div>
    </header>
    <div
      class="flex"
    >
      <div
        class="flex items-center pr-2"
      >
        <img
          class="p-0 m-0 w-12 h-12 rounded-full cursor-pointer"
          src="testurl"
        />
      </div>
      <div
        class="pl-2"
      >
        <div>
          headline_1 
          <a
            class="underline text-pink-700"
            href="githuburl"
            rel="noreferrer"
            target="_blank"
          >
            test
          </a>
        </div>
        <div>
          headline_2
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
